test/cluster/mzcompose.py

```diff
@@ -2546,3 +2547,68 @@ def worker(c: Composition, i: int) -> None:
         assert (
             p99 < p99_limit
         ), f"p99 is {p99:.2f}s, should be less than {p99_limit:.2f}s"
+
+
+def workflow_test_incident_70(c: Composition) -> None:
+    """
+    Test incident-70.
+    """
+    num_conns = 10
+    persist_reader_lease_duration_in_sec = 30
+    data_scale_factor = 10
+
+    c.down(destroy_volumes=True)
+    c.up("materialized")
+
+    c.sql(
+        f"ALTER SYSTEM SET max_connections = {num_conns + 5};",
+        port=6877,
+        user="mz_system",
+    )
+
+    c.sql(
+        f"ALTER SYSTEM SET persist_reader_lease_duration = '{persist_reader_lease_duration_in_sec}s';",
+        port=6877,
+        user="mz_system",
+    )
+
+    c.sql(
+        f"""
+        CREATE SOURCE gen FROM LOAD GENERATOR TPCH (SCALE FACTOR {data_scale_factor}, TICK INTERVAL '100ms') FOR ALL TABLES;
+
+        CREATE MATERIALIZED VIEW mv_lineitem_count AS SELECT count(*) FROM lineitem;
+        """
+    )
+
+    start_time = datetime.now()
+    end_time = start_time + timedelta(
+        seconds=persist_reader_lease_duration_in_sec * 3 + 10
+    )
+
+    def worker(c: Composition, worker_index: int) -> None:
+        print(f"Thread {worker_index} tries to acquire a cursor")
+        cursor = c.sql_cursor()
+        print(f"Thread {worker_index} got a cursor")
+
+        iteration = 1
+        while datetime.now() < end_time:
+            if iteration % 20 == 0:
+                print(f"Thread {worker_index}, iteration {iteration}")
+            cursor.execute("SELECT * FROM mv_lineitem_count;")
```

Comment:
I also didn't have any ad-hoc queries, just the MVs. I don't have any reason to think that the ad-hoc queries would drive load less effectively, but all the CPU during that incident was being spent rehydrating existing dataflows.